### PR TITLE
Wait for genesis when starting node

### LINKED ISSUE #991: Wait for genesis when starting node
If genesis time hasn't come yet, wait for genesis before starting client.

### PR DESCRIPTION
**Pull request by cyberbono3**
PR resolves #991

Should I handle the error manually let std_duration = duration.to_std()?  in wait_until_genesis in order to return Error instead of OutOfRangeError?

### CODE REVIEW

File: chain/client/src/client.rs [hunk 1]
```diff
@@ -76,7 +89,13 @@ impl ClientActor {
         network_actor: Recipient<NetworkRequests>,
         block_producer: Option<BlockProducer>,
     ) -> Result<Self, Error> {
+
         // TODO: Wait until genesis.
+
+         //think about it
+        let _ = wait_until_genesis(&genesis_time)?;
```
Comment (ilblackdragon) [inline chain/client/src/client.rs]: don't need `let _ =`
and think about it -> Waiting until genesis is started early.

File: chain/client/src/client.rs [hunk 2]
```diff
@@ -76,7 +89,13 @@ impl ClientActor {
         network_actor: Recipient<NetworkRequests>,
         block_producer: Option<BlockProducer>,
     ) -> Result<Self, Error> {
+
         // TODO: Wait until genesis.
```
Comment (ilblackdragon) [inline chain/client/src/client.rs]: don't need todo anymore

File: chain/client/src/client.rs [hunk 3]
```diff
@@ -65,6 +68,16 @@ pub struct ClientActor {
...
     /// Total number of transactions processed.
     num_tx_processed: u64,
+
+
+}
+
+fn wait_until_genesis(genesis_time: &DateTime<utc>) -> Result<(), OutOfRangeError> {
+        let now : DateTime<Utc> = Utc::now();
+        let duration = now.signed_duration_since(*genesis_time);
+        let std_duration = duration.to_std()?;
```
Comment (ilblackdragon) [inline chain/client/src/client.rs]: hm... not actually sure what's the best way here, may be `Duration::from_seconds(duration.seconds())` ?

File: chain/client/src/client.rs [hunk 4]
```diff
@@ -65,6 +68,16 @@ pub struct ClientActor {
     num_blocks_processed: u64,
     /// Total number of transactions processed.
     num_tx_processed: u64,
+
```
Comment (ilblackdragon) [inline chain/client/src/client.rs]: don't need extra space here

File: chain/client/src/client.rs [hunk 5]
```diff
@@ -65,6 +68,16 @@ pub struct ClientActor {
...
+
+}
+
+fn wait_until_genesis(genesis_time: &DateTime<utc>) -> Result<(), OutOfRangeError> {
+        let now : DateTime<Utc> = Utc::now();
+        let duration = now.signed_duration_since(*genesis_time);
+        let std_duration = duration.to_std()?;
+        thread::sleep(std_duration);
```
Comment (ilblackdragon) [inline chain/client/src/client.rs]: let's actually add logging here target: "info", "Waiting for genesis: " and pretty printed time.

Comment (cyberbono3) [reply chain/client/src/client.rs]: what logger has to be used? There are init_integration_logger ,  init_test_logger, env_logger available.

File: chain/client/src/client.rs (hunk 3 above)
Comment (cyberbono3) [reply chain/client/src/client.rs]: I think,  chrono::Duration has to be converted to std::time::Duration in order to be passed to thread::sleep as a parameter.
to_std() method actually resolves this issue.

File: chain/client/src/client.rs (hunk 1 above)
Comment (ilblackdragon) [reply chain/client/src/client.rs]: nvm, you do need that.

File: chain/client/src/client.rs (hunk 5 above)
Comment (ilblackdragon) [reply chain/client/src/client.rs]: you don't need to use those, that is for setting up logging, done once per program -> just use `info!(target: "info", "...")`

File: chain/client/src/client.rs (hunk 3 above)
Comment (ilblackdragon) [reply chain/client/src/client.rs]: From my understanding `Duration::from_seconds(duration.seconds())` does the same and doesn't raise error. Am I wrong?

**Pull request closed without merging**